File: 0125-valid-palindrome/solution.swift
class Solution {
    func isPalindrome(_ s: String) -> Bool {
        var n = s.count
        var i=0
        var j=n-1

        var chars = Array(s)

        while(i < j) {
            if !chars[i].isLetter && !chars[i].isNumber {
                i += 1
                continue
            } 
            
            if !chars[j].isLetter && !chars[j].isNumber {
                j -= 1
                continue
            }

            if chars[i].lowercased() != chars[j].lowercased() {
                return false
            }

                i += 1
                j -= 1
        }
        return true
    }
}

File: 0125-valid-palindrome/solution.java
class Solution {
    public boolean isPalindrome(String s) {
        //if we find non alphanumeric character at i, then i++, if at j then j--
        int n = s.length();
        int i=0;
        int j=n-1;

        while(i < j) {
            char ch1 = s.charAt(i);
            char ch2 = s.charAt(j);

            if(!Character.isLetter(ch1) && !Character.isDigit(ch1)) {
                i++;
                continue;
            }
            if(!Character.isLetter(ch2) && !Character.isDigit(ch2)) {
                j--;
                continue;
            } 
                if(Character.toLowerCase(ch1) == Character.toLowerCase(ch2)) {
                    i++;
                    j--;
                } else {
                    return false;
                }
            
        }
        return true;
    }
}
